jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletHandler.java

```diff
@@ -452,17 +455,18 @@ public void doScope(String target, Request baseRequest, HttpServletRequest reque
                 String servletPath = pathSpec.getPathMatch(target);
                 String pathInfo = pathSpec.getPathInfo(target);
 
+                HttpServletMapping httpServletMapping = mapping.getHttpServletMapping(servletPath);
                 if (DispatcherType.INCLUDE.equals(type))
                 {
                     baseRequest.setAttribute(RequestDispatcher.INCLUDE_SERVLET_PATH, servletPath);
                     baseRequest.setAttribute(RequestDispatcher.INCLUDE_PATH_INFO, pathInfo);
-                    baseRequest.setAttribute(RequestDispatcher.INCLUDE_MAPPING, Request.getServletMapping(pathSpec, servletPath, servletHolder.getName()));
+                    baseRequest.setAttribute(RequestDispatcher.INCLUDE_MAPPING, httpServletMapping);
                 }
                 else
                 {
-                    baseRequest.setPathSpec(pathSpec);
                     baseRequest.setServletPath(servletPath);
                     baseRequest.setPathInfo(pathInfo);
+                    baseRequest.setHttpServletMapping(httpServletMapping);
                 }
             }
         }
```

Comment:
in the log debug below, maybe just log the mapping and make sure it has a good toString